### PR TITLE
[11.0][stock_no_negative][IMP] Add the ability to allow negative stock for individual stock locations.

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
With this PR you can set the option "Allow negative stock" for individual locations.

#. Go to *Inventory / Configuration / Settings* and activate
   the option *Storage Locations*.
#. Go to *Inventory / Configuration / Warehouse Management / Locations* and
   activate the option the option *Allow Negative Stock* for the locations you
   choose.

### CODE REVIEW

**Review by etobella — APPROVED** [review]
Great! That is a better approach than my PR #480

**Comment by rousseldenis** [comment]
@jbeficent Check travis

**Comment by JordiBForgeFlow** [comment]
@rousseldenis Done!

@alexis-via As this one is your baby, what do you think of the proposal?

File: stock_no_negative/models/stock_location.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from odoo import models, fields
+
+
+class StockLocation(models.Model):
+    _inherit = 'stock.location'
+
+    allow_negative_stock = fields.Boolean(
+        string='Allow Negative Stock',
+        help="Allow negative stock levels for the stockable products "
+        "attached to this location. The options doesn't apply to products "
```
Comment (alexis-via) [inline stock_no_negative/models/stock_location.py]: I cannot understand the second sentence of the help message.

**Review by alexis-via — APPROVED** [review]

File: stock_no_negative/models/stock_location.py (hunk 1 above)
Comment (pedrobaeza) [reply stock_no_negative/models/stock_location.py]: @jbeficent what about this?

Comment (pedrobaeza) [reply stock_no_negative/models/stock_location.py]: Any news about this?

Comment (JordiBForgeFlow) [reply stock_no_negative/models/stock_location.py]: @pedrobaeza @alexis-via I have fixed this last sentence that was wrong.

**Comment by JordiBForgeFlow** [comment]
Merging, as it is now approved.

**Pull request merged**